### ISSUE TITLE
Linux Target build not working from Mac

### ISSUE THREAD

**Issue by mrgodhani** [electron-userland/electron-builder#3440]
* **Version**: 20.31.2

* **Target**: Linux

I am trying to build for Linux. I am seeing this error constantly. Following below is the log:

```
• application Linux category is set to default "Utility" reason=linux.category is not set and cannot map from macOS docs=https://electron.build/configuration/configuration#LinuxBuildOptions-category
  • application Linux category is set to default "Utility" reason=linux.category is not set and cannot map from macOS docs=https://electron.build/configuration/configuration#LinuxBuildOptions-category
  ⨯ expected argument for flag '--executable'

  ⨯ error: exit status 1
path: /Users/<user>/Library/Caches/electron-builder/appimage/appimage-9.1.0/darwin/mksquashfs
args: [/Users/<user>/Library/Caches/electron-builder/appimage/appimage-9.1.0/darwin/mksquashfs /Users/<user>/Desktop/WebProjects/rss-reader/build/__appImage-x64 /Users/<user>/Desktop/WebProjects/rss-reader/build/@mrgodhani/rss-reader-0.3.6-x86_64.AppImage -offset 100024 -all-root -noappend -no-progress -quiet -no-xattrs -no-fragments]
output: Could not create destination file: No such file or directory

github.com/develar/app-builder/pkg/util.Execute
	/Users/develar/go/src/github.com/develar/app-builder/pkg/util/util.go:66
github.com/develar/app-builder/pkg/appimage.createSquashFs
	/Users/develar/go/src/github.com/develar/app-builder/pkg/appimage/appImage.go:183
github.com/develar/app-builder/pkg/appimage.AppImage
	/Users/develar/go/src/github.com/develar/app-builder/pkg/appimage/appImage.go:126
github.com/develar/app-builder/pkg/appimage.ConfigureCommand.func1
	/Users/develar/go/src/github.com/develar/app-builder/pkg/appimage/appImage.go:72
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*actionMixin).applyActions
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/actions.go:28
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).applyActions
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:557
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).execute
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:390
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).Parse
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:222
main.main
	/Users/develar/go/src/github.com/develar/app-builder/main.go:74
runtime.main
	/usr/local/Cellar/go/1.11.1/libexec/src/runtime/proc.go:201
runtime.goexit
	/usr/local/Cellar/go/1.11.1/libexec/src/runtime/asm_amd64.s:1333
github.com/develar/app-builder/pkg/appimage.createSquashFs
	/Users/develar/go/src/github.com/develar/app-builder/pkg/appimage/appImage.go:185
github.com/develar/app-builder/pkg/appimage.AppImage
	/Users/develar/go/src/github.com/develar/app-builder/pkg/appimage/appImage.go:126
github.com/develar/app-builder/pkg/appimage.ConfigureCommand.func1
	/Users/develar/go/src/github.com/develar/app-builder/pkg/appimage/appImage.go:72
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*actionMixin).applyActions
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/actions.go:28
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).applyActions
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:557
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).execute
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:390
github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin.(*Application).Parse
	/Users/develar/go/src/github.com/develar/app-builder/vendor/github.com/alecthomas/kingpin/app.go:222
main.main
	/Users/develar/go/src/github.com/develar/app-builder/main.go:74
runtime.main
	/usr/local/Cellar/go/1.11.1/libexec/src/runtime/proc.go:201
runtime.goexit
	/usr/local/Cellar/go/1.11.1/libexec/src/runtime/asm_amd64.s:1333

Error: Cannot cleanup:

Error #1 --------------------------------------------------------------------------------
Error: /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-bin/mac/app-builder exited with code 1
    at ChildProcess.childProcess.once.code (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/util.ts:253:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
From previous event:
    at SnapTarget.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/targets/snap.ts:146:44)
    at Function.buildAsyncTargets (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:139:36)
    at LinuxPackager.packageInDistributableFormat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:118:24)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:113:10
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
From previous event:
    at LinuxPackager.pack (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:110:95)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:375:24
    at Generator.next (<anonymous>)
    at xfs.stat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqCallback.oncomplete (fs.js:162:5)
From previous event:
    at Packager.doBuild (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:343:39)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:313:57
    at Generator.next (<anonymous>)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
From previous event:
    at Packager._build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:284:133)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:280:23
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
From previous event:
    at Packager.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:237:14)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/index.ts:58:28)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/builder.ts:227:10)
    at then (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/cli/cli.ts:46:19)

Error #2 --------------------------------------------------------------------------------
Error: /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-bin/mac/app-builder exited with code 1
    at ChildProcess.childProcess.once.code (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/util.ts:253:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
From previous event:
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
From previous event:
    at AppImageTarget.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/targets/AppImageTarget.ts:28:44)
    at Function.buildAsyncTargets (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:139:36)
    at LinuxPackager.packageInDistributableFormat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:118:24)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:113:10
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
From previous event:
    at LinuxPackager.pack (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:110:95)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:375:24
    at Generator.next (<anonymous>)
    at xfs.stat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqCallback.oncomplete (fs.js:162:5)
From previous event:
    at Packager.doBuild (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:343:39)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:313:57
    at Generator.next (<anonymous>)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
From previous event:
    at Packager._build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:284:133)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:280:23
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
From previous event:
    at Packager.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:237:14)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/index.ts:58:28)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/builder.ts:227:10)
    at then (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/cli/cli.ts:46:19)
    at ChildProcess.childProcess.once.code (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/util.ts:253:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
    at SnapTarget.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/targets/snap.ts:146:44)
    at Function.buildAsyncTargets (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:139:36)
    at LinuxPackager.packageInDistributableFormat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:118:24)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:113:10
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
    at LinuxPackager.pack (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:110:95)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:375:24
    at Generator.next (<anonymous>)
    at xfs.stat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqCallback.oncomplete (fs.js:162:5)
    at Packager.doBuild (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:343:39)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:313:57
    at Generator.next (<anonymous>)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
    at Packager._build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:284:133)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:280:23
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
    at Packager.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:237:14)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/index.ts:58:28)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/builder.ts:227:10)
    at then (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/cli/cli.ts:46:19)
    at ChildProcess.childProcess.once.code (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/util.ts:253:14)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at maybeClose (internal/child_process.js:962:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:251:5)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
    at AppImageTarget.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/targets/AppImageTarget.ts:28:44)
    at Function.buildAsyncTargets (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:139:36)
    at LinuxPackager.packageInDistributableFormat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:118:24)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:113:10
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
    at LinuxPackager.pack (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/platformPackager.ts:110:95)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:375:24
    at Generator.next (<anonymous>)
    at xfs.stat (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/fs-extra/lib/mkdirs/mkdirs.js:56:16)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/polyfills.js:287:18
    at FSReqCallback.oncomplete (fs.js:162:5)
    at Packager.doBuild (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:343:39)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:313:57
    at Generator.next (<anonymous>)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
    at Packager._build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:284:133)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:280:23
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
    at Packager.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:237:14)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/index.ts:58:28)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/builder.ts:227:10)
    at then (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/cli/cli.ts:46:19)
    at throwError (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/asyncTaskManager.ts:90:11)
    at checkErrors (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/asyncTaskManager.ts:53:9)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/asyncTaskManager.ts:67:7
From previous event:
    at AsyncTaskManager.awaitTasks (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/builder-util/src/asyncTaskManager.ts:44:19)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:387:23
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
From previous event:
    at Packager.doBuild (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:343:39)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:313:57
    at Generator.next (<anonymous>)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:99:16
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/graceful-fs/graceful-fs.js:43:10
    at FSReqCallback.oncomplete (fs.js:148:20)
From previous event:
    at Packager._build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:284:133)
    at /Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:280:23
    at Generator.next (<anonymous>)
    at processImmediate (timers.js:632:19)
From previous event:
    at Packager.build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/packager.ts:237:14)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/app-builder-lib/src/index.ts:58:28)
    at build (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/builder.ts:227:10)
    at then (/Users/<user>/Desktop/WebProjects/rss-reader/node_modules/electron-builder/src/cli/cli.ts:46:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
any help how to resolve this ? Or am I missing anything ? Even though if I assign category I would still get `Could not create destination file: No such file or directory` error.

**Comment by develar** [comment]
What is the name in package.json? Maybe problem is that path contains `@` (@mrgodhani/rss-reader).

**Comment by mrgodhani** [comment]
@develar Yes that's the name of the package. I will try to change the package name and try to build again. But this only happens for Linux FYI works well for other platform

**Comment by mrgodhani** [comment]
@develar You are right that fixes it. Thank you !

**Comment by artdiniz** [comment]
I'm getting this error too. 

• **Version**: 20.38.5
• **Target**: Linux

Removing \@scope from package name fixes it, but leaving the package name without \@scope is not desirable right now. 

Tried to change `executableName`, and resulted in error:
```
Could not create destination file: No such file or directory
```

Is there any config I can set to get over this?

**Comment by artdiniz** [comment]
**TL;DR;**

Could get over setting this is in `package.json` `build` config:

```
"linux": {
    "executableName": "cake",
    "artifactName": "${productName}-${version}.${ext}"
}
```

___

Found out that electron-builder wasn't using the `productName` set in `package.json` (only for the --linux build).

Changing  `executableName` was halfway of fixing this. I also had to change `artifactName`. 

I followed the documentation that says the default `artifactName` is `${productName}-${version}.${ext}`. It says that it can change a bit from platform to platform but I didn't found anything about what is being used as `artifactName` for linux builds. Changed to the documentation default and everything works. 

Are `artifactName ` and `executableName` ignoring `productName`? If so, should that be fixed or maybe written in the docs?

**Comment by maneetgoyal** [comment]
Was facing same issue while trying to create Linux distributions on Windows via WSL. The above solution by @artdiniz 🎉:

```
"linux": {
    "executableName": "desiredName",
    "artifactName": "${productName}-${version}.${ext}"
}
```

fixed it.
